addons/purchase/models/product.py

```diff
@@ -19,11 +19,21 @@ class ProductTemplate(models.Model):
     purchase_method = fields.Selection([
         ('purchase', 'On ordered quantities'),
         ('receive', 'On received quantities'),
-    ], string="Control Policy", help="On ordered quantities: Control bills based on ordered quantities.\n"
-        "On received quantities: Control bills based on received quantities.", default="receive")
+    ], string="Control Policy", compute='_compute_purchase_method', default='receive', store=True, readonly=False,
```

Comment:
@Whenrow @ticodoo 
Shouldn't the default be a lambda here, with a default method with the same behaviour as the compute?
I guess we'd like service products to have a 'purchase' method by default.

Reply:
Hello @AH-Yussef, vava is correct. If the product is directly created (e.g. through an xml-rpc/another method creates it directly) or when there is a `'default_detailed_type': 'service'` used in a view for newly created products, then only the default 'receive' will be assigned and the compute won't be called unless the user changes the `detailed_type`. It is better to change the default to a lambda for consistent behavior in these cases.